### PR TITLE
Add using offline token login method

### PR DESCRIPTION
**Pull request by Phaow**
## Add login method with offline token
This is an enhancement for login rh-sso kcp service with an offline token. It could be helpful when we execute the kcp tests in CI environments. We could use the method login the kcp service without open a browser input the username and password. 

**Test Record**
```console
wangpenghao@MacBook-Pro  ~/Upstream/kcp-tests   login-enhance  source bin/set-offline-token                    
 wangpenghao@MacBook-Pro  ~/Upstream/kcp-tests   login-enhance  make login
./hack/login_with_offline_token.sh
INFO: Log into kcp service with "OFFLINE_TOKEN" succeed
 wangpenghao@MacBook-Pro  ~/Upstream/kcp-tests   login-enhance  k get ws
NAME              TYPE           PHASE   URL
rh-sso-XXXXXX   organization   Ready   https://<domain>/clusters/root:rh-sso-XXXXXX
```

### CODE REVIEW

**Comment by Phaow** [comment]
@xingxingxia @kasturinarra @wangke19 @duanwei33 @zxiao-redhat Could you please help to have a review when you get a chance? Thanks!

**Comment by kasturinarra** [comment]
@Phaow i understand that the instructions to run the cases would change as well, i.e user who would want to run the tests would need to export their token before the run starts, if my understanding is right, could you please help add the modified steps along with this PR for review ? Thanks !!

**Comment by Phaow** [comment]
> @Phaow i understand that the instructions to run the cases would change as well, i.e user who would want to run the tests would need to export their token before the run starts, if my understanding is right, could you please help add the modified steps along with this PR for review ? Thanks !!
Hi @kasturinarra ,sure, BTW, using offline token login method only works for the kcp deploy with redhat sso, so it maybe not a common way to trigger the test. I'm not very sure we should add it to the `README.md`, maybe add it to a interal doc is better. WDYT?

**Comment by kasturinarra** [comment]
> > @Phaow i understand that the instructions to run the cases would change as well, i.e user who would want to run the tests would need to export their token before the run starts, if my understanding is right, could you please help add the modified steps along with this PR for review ? Thanks !!
> > Hi @kasturinarra ,sure, BTW, using offline token login method only works for the kcp deploy with redhat sso, so it maybe not a common way to trigger the test. I'm not very sure we should add it to the `README.md`, maybe add it to a interal doc is better. WDYT?

@Phaow IMO some one looking at the repo should be able to run all the cases present in their with out any fail and the instructions should be straight forward. We could say that if kcp is not deployed with RH SSO, please skip this section and go to next one ? WDYS ?

**Comment by Phaow** [comment]
> > > @Phaow i understand that the instructions to run the cases would change as well, i.e user who would want to run the tests would need to export their token before the run starts, if my understanding is right, could you please help add the modified steps along with this PR for review ? Thanks !!
> > > Hi @kasturinarra ,sure, BTW, using offline token login method only works for the kcp deploy with redhat sso, so it maybe not a common way to trigger the test. I'm not very sure we should add it to the `README.md`, maybe add it to a interal doc is better. WDYT?
> 
> @Phaow IMO some one looking at the repo should be able to run all the cases present in their with out any fail and the instructions should be straight forward. We could say that if kcp is not deployed with RH SSO, please skip this section and go to next one ? WDYS ?

Sounds good. After the https://github.com/kcp-dev/kcp-tests/pull/20 merged, I'll add the introduction or there will have merge conflict. Or we could add the instructions in a separate pull request, both ways should be ok.

**Comment by Phaow** [comment]
Hi @kasturinarra ,update introduction and rebase the latest code, could you please help take a look when you get a chance? Thanks!

File: README.md [hunk 1]
```diff
@@ -25,6 +25,11 @@ This repository holds the kcp tests that tests against the publicly available  i
   ```console
   $ kubectl oidc-login get-token --oidc-issuer-url=<oidc issuer url> --oidc-client-id=<oidc client ID> --oidc-redirect-url-hostname=127.0.0.1
   ```
+  - For kcp services deploy with redhat sso( **Only supported kcp services deploy with redhat sso** ), there's another login method by use `offline token`, you could get an offline token ( [How to generate an offline token](https://access.redhat.com/articles/3626371#bgenerating-a-new-offline-tokenb-3) ) and log in to the kcp service by execute the follow commands.
+    ```console
+    $ export OFFLINE_TOKEN=<YOUR_OFFLINE_TOKEN>
+    $ make login
+    ```
```
Comment (kasturinarra) [inline README.md:34]: @Phaow or should we say as below ?
please make sure to export OFFLINE_TOKEN and run make login before running the smoke tests if kcp services have been deployed with redhat SSO ?

Comment (Phaow) [reply README.md:34]: Hmm, from my understanding we support both 2 ways use offline token or previous way `username/password`, so it's not a must step for running kcp smoke test just a convenient way to login without open a browser and input the `username/password`. So I think we seems no need to add the point. WDYT?

File: README.md [hunk 2]
```diff
@@ -25,6 +25,11 @@ This repository holds the kcp tests that tests against the publicly available  i
   ```console
   $ kubectl oidc-login get-token --oidc-issuer-url=<oidc issuer url> --oidc-client-id=<oidc client ID> --oidc-redirect-url-hostname=127.0.0.1
   ```
+  - For kcp services deploy with redhat sso( **Only supported kcp services deploy with redhat sso** ), there's another login method by use `offline token`, you could get an offline token ( [How to generate an offline token](https://access.redhat.com/articles/3626371#bgenerating-a-new-offline-tokenb-3) ) and log in to the kcp service by execute the follow commands.
```
Comment (kasturinarra) [inline README.md]: s/deploy/deployed
Do not think we need that text in Bold Letters
s/there's another login method by use/there's another login method called `offline token` that could be used.
s/execute the follow commands/executing the following commands with out actually needing to input credentials into the browser.

File: README.md (hunk 1 above)
Comment (kasturinarra) [reply README.md:34]: @Phaow make sense and agree, Just added few minor comments and once addressed we are good to be merged, thanks !!

File: README.md (hunk 2 above)
Comment (Phaow) [reply README.md]: Hmm, bold Letters may be not necessary  and update the descriptions, addressed.

File: README.md [hunk 3]
```diff
@@ -25,7 +25,7 @@ This repository holds the kcp tests that tests against the publicly available  i
   ```console
   $ kubectl oidc-login get-token --oidc-issuer-url=<oidc issuer url> --oidc-client-id=<oidc client ID> --oidc-redirect-url-hostname=127.0.0.1
   ```
-  - For kcp services deploy with redhat sso( **Only supported kcp services deploy with redhat sso** ), there's another login method by use `offline token`, you could get an offline token ( [How to generate an offline token](https://access.redhat.com/articles/3626371#bgenerating-a-new-offline-tokenb-3) ) and log in to the kcp service by execute the follow commands.
+  - For kcp services deploy with redhat sso( **Only supported kcp services deploy with redhat sso** ), there's another login method called `login with offline token` that could be used, you could get an offline token ( [How to generate an offline token](https://access.redhat.com/articles/3626371#bgenerating-a-new-offline-tokenb-3) ) and log in to the kcp service by executing the following commands without inputting credentials into the browser.
```
Comment (kasturinarra) [inline README.md]: @Phaow still see text in the bold here, also s/without inputting/ with out having to input

Comment (Phaow) [reply README.md]: Sorry, I forgot update it, addressed.

Comment (kasturinarra) [reply README.md]: Thanks, @Phaow sorry if my suggestion was not clear, what i meant was to remove the text in bold completely.

Comment (Phaow) [reply README.md]: np, addressed, I misunderstand.

**Pull request merged**